### ISSUE TITLE
Validation of nested UserControls does not work

### ISSUE THREAD

**Issue by WolfgangHG** [dotnet/winforms#8457]
### .NET version

.NET6

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

I think the problem also occured in e.g. NET Framework 4.8.

### Issue description

Attached sample has a TextBox on a UserControl. The UserControl.OnValidate override shows a validation error.
Another Button is placed on a UserControl inside of a UserControl. 

When first focusing the textbox, then clicking on the button in the nested user control, the validation error is shown the first time. But the next button click works, which is not the expected behavior, as the validation error should be shown again and the button should not be focused.

### Steps to reproduce

See sample:
[UserControlValidation.zip](https://github.com/dotnet/winforms/files/10351263/UserControlValidation.zip)

The bottom textbox (on orange user control) and the bottom button (green user control on blue user control) reproduce the problem: focus the textbox, then click the button marked in the screenshot => validation error is shown. Click the button again => the validation error is not shown, but the button click handler is raised.
It works if you click any other (non-nested) button on the form. And it works if you click the first "Validating TextBox", which is not nested.
![validation](https://user-images.githubusercontent.com/138049/210756596-1337222e-0321-4d5c-9895-98b4f138aa65.png)

**Comment by merriemcgaw** [comment]
Thank you for reporting this issue! After we've analyzed, we determined a fix here could be breaking for other customer apps that are dealing with this issue. In many cases un-nesting the controls is the right way to work around this. Given our current priorities, and the fact that it has always behaved this way, we are going to close this as won't fix. If we hear significant customer feedback requesting this feature, we may be able to reopen this in the future.
